### ISSUE TITLE
F3 3.6 and PHP7.1.2 splits session data

### ISSUE THREAD

**Issue by module0x90** [bcosca/fatfree-core#196]
Hi,

I have just replaced F3 3.5 with F3 3.6.
When I try to login (using DB based sessions), it just kicks me back out. Sessions data are created, more on that later.

I found https://groups.google.com/forum/#!topic/f3-framework/WfCJpW479Vw and applied the changes from 4f50cd0aa and 99b0dbdd5, but only for db/sql/session.php and session.php.

My session table looks like this:

```
*************************** 1. row ***************************
session_id: 0f3788b0be33d6ff0f0b49bf1c2d2ad7
      data: LastPageURL|s:0:"";CurrentPageURL|s:6:"/login";flash|a:2:{s:3:"msg";a:0:{}s:3:"key";N;}
        ip: XXXX
     agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36
     stamp: 1487870281
      csrf:
*************************** 2. row ***************************
session_id: 1ebea216ad18349e9303cc532df8b481
      data: user|s:6:"tomtom";
        ip: XXXX
     agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36
     stamp: 1487870281
      csrf:
*************************** 3. row ***************************
session_id: f6171995f7c94a0a554e38e288f83a9a
      data: LastPageURL|s:0:"";CurrentPageURL|s:1:"/";
        ip: XXXX
     agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36
     stamp: 1487870281
      csrf:
```

It appears to me for every **$f3->set('SESSION.** the session id gets regenerated and it kinda splits up the vars into different sessions. The browser cookie I got is **0f3788b0be33d6ff0f0b49bf1c2d2ad7**, and that isn't referring **user: tomtom** - so no valid session and it requires to log me back in.
This was all working with 3.5 :-)

Cheers

Thomas

PS: I tried -dev, but oi! that changed too much.

**Comment by Rayne** [comment]
> I tried -dev, but oi! that changed too much.

This sound like you were testing `fatfree` with the `master` and `dev` branch (which shouldn't be used for websites as it contains a unit testing website itself). Did you also test the `master` branch of `fatfree-core` ([bcosca/fatfree-core@Packagist](https://packagist.org/packages/bcosca/fatfree-core))?

**Comment by module0x90** [comment]
I was using **composer require bcosca/fatfree-core** which pulled in 3.6. 
I "git reset" back to 3.6 safely after trying **composer require bcosca/fatfree-core:dev-master**.

**Comment by module0x90** [comment]
Seems to have to do something with my NginX config on my own -dev environment.

**Comment by module0x90** [comment]
Ok, I found now the problem. Just documenting this for posterity.

Problem in a nutshell: Logging in succeeded, but when it accessed the session the ID has changed again and it threw me back at /login.

I got from https://groups.google.com/forum/#!searchin/f3-framework/session%7Csort:relevance/f3-framework/p-DEx0e0XpY/6NvncjhuCAAJ the idea why I am having this problem:

I am using CSRF tokens on POST forms, and also Dice as a depinj - for injecting \DB\SQL\Session into the app. Now in my setup, I am not doing CSRF checks on login, so my fancy csrf() function isn't called,
which then would access the SESSION.user, as this never happens, the session object isn't instantiated -> session is empty.
As a workaround, I helped myself in accessing SESSION.user manually after the session object is stored, so this makes it to get instantiated at all times.
